src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -185,6 +190,15 @@ public Assembly LoadFromAssemblyName(AssemblyName assemblyName)
             Assembly assembly = Load(assemblyName);
             if (assembly == null)
             {
+                // Invoke the AssemblyResolve event callbacks if wired up
+                if (AssemblyResolve != null)
```

Comment:
The null check is unnecessary since it is done in GetFirstResolvedAssembly as well. Also, I think you may still need to cache it in the local for the null check to avoid races.